src/server/object_services/object_server.js

```diff
@@ -1591,6 +1591,44 @@ function check_md_conditions(conditions, obj) {
     }
 }
 
+/**
+ * Return the etag ("Entity tag") for the given entity.
+ * Entity can be ObjectMD or ObjectMultipart or an updates for one of those.
+ * 
+ * Notice that if the etag field is returns from md5 hex then we can put it as is,
+ * however if we use a sha256 or id we have to add some prefix with a dash so that 
+ * s3 clients can understand that this is not an md5.
+ * 
+ * These fallbacks allow us to configure our endpoints to disable md5 calculations
+ * for use cases where performance matters more, see config.IO_CALC_MD5_ENABLED.
+ * 
+ * @typedef {{
+ *  etag?: string;
+ *  md5_b64?: string;
+ *  sha256_b64?: string;
+ *  _id?: nb.ID;
+ * }} EtagEntity
+ *
+ * @param {EtagEntity} entity
+ * @param {EtagEntity} [updates]
+ * @returns {string}
+ */
+function get_etag(entity, updates) {
+   const etag = updates?.etag || entity.etag;
+   if (etag) return etag;
+
+   const md5_b64 = updates?.md5_b64 || entity.md5_b64;
+   if (md5_b64) return Buffer.from(md5_b64, 'base64').toString('hex');
+
+   const sha256_b64 = updates?.sha256_b64 || entity.sha256_b64;
+   if (sha256_b64) return 'sha256-' + Buffer.from(sha256_b64, 'base64').toString('hex');
+
+   const id = updates?._id || entity._id;
+   if (id) return 'id-' + id.toHexString();
+
+   return '';
```

Comment:
@romayalon I wonder if I should throw here instead of silently "failing" ... WDYT?

Reply:
Is there a situation where we get there? I couldn't find one and I thought that id is the default

Reply:
right, we shouldn't get there, this is why I thought it might be better to throw

Reply:
right, we shouldn't get there, this is why I thought it might be better to throw